Source/cryptalgo/SecureSocketPort.cpp

```diff
@@ -131,36 +131,46 @@ SecureSocketPort::Handler::~Handler() {
 }
 
 uint32_t SecureSocketPort::Handler::Initialize() {
-    uint32_t success = Core::ERROR_NONE;
-
-    _context = SSL_CTX_new(TLS_method());
-
-    _ssl = SSL_new(static_cast<SSL_CTX*>(_context));
-    SSL_set_fd(static_cast<SSL*>(_ssl), static_cast<Core::IResource&>(*this).Descriptor());
-    SSL_CTX_set_options(static_cast<SSL_CTX*>(_context), SSL_OP_ALL | SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3);
-
-    // Trust the same certificates as any other application
-    if (SSL_CTX_set_default_verify_paths(static_cast<SSL_CTX*>(_context)) == 1) {
-        success = Core::SocketPort::Initialize();
-    } else {
-        TRACE_L1("OpenSSL failed to load certificate store");
-        success = Core::ERROR_GENERAL;
+    uint32_t success = Core::ERROR_GENERAL;
+
+    if ((_context = SSL_CTX_new(TLS_method())) != nullptr) {
+        if (   // Returns bit-mask after adding options
+               ((SSL_CTX_set_options(static_cast<SSL_CTX*>(_context), SSL_OP_ALL | SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3) & (SSL_OP_ALL | SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3))  == (SSL_OP_ALL | SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3))
```

Comment:
The SSL_CTX_set_options contruction feels a bit like you are checking the openSSL implementation if it does what you are asking it to do. The manual does not give any restrictions on mutal exclusive options. So since we do not actively want to check third party libraries in Production or Release I suggest putting this check in an ASSERT if you want to check OpenSSL